r/R/dplyr-functions.R

```diff
@@ -848,6 +844,21 @@ nse_funcs$wday <- function(x,
   Expression$create("day_of_week", x, options = list(count_from_zero = FALSE, week_start = week_start))
 }
 
+nse_funcs$is.Date <- function(x) {
+  inherits(x, "Date") ||
+    (inherits(x, "Expression") && x$type_id() %in% Type[c("DATE32")])
```

Comment:
I think we want `DATE64` in here as well since those are treated as dates in other areas (even though they do have a bit funny millisecond precision). Possibly also below for `is.instant()` too?